agent/consul/catalog_endpoint.go

```diff
@@ -284,13 +286,17 @@ func (c *Catalog) ListNodes(args *structs.DCSpecificRequest, reply *structs.Inde
 		})
 }
 
+func isUnmodified(opts structs.QueryOptions, index uint64) bool {
+	return opts.AllowNotModifiedResponse && opts.MinQueryIndex > 0 && opts.MinQueryIndex == index
+}
+
 // ListServices is used to query the services in a DC
 func (c *Catalog) ListServices(args *structs.DCSpecificRequest, reply *structs.IndexedServices) error {
 	if done, err := c.srv.forward("Catalog.ListServices", args, args, reply); done {
 		return err
 	}
 
-	(*reply).EnterpriseMeta = args.EnterpriseMeta
+	reply.EnterpriseMeta = args.EnterpriseMeta
```

Comment:
I think the dereference might be for tests when this is not a real servers (as many tests have issues with inmemdb because the value provided is modified in a async way in the test), I had similar issue not with EntrepriseMeta, but with other parameters in queries, for instance in this PR: https://github.com/hashicorp/consul/pull/7612
It might have been done a a kind of defensive practice for such tests